src/libs/actions/Lounge.js

```diff
@@ -0,0 +1,29 @@
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+
+/**
+ * Register visit at the lounge
+ *
+ * @param {String} checkInsRemaining
+ */
+function recordLoungeVisit(checkInsRemaining) {
+    API.write(
+        'RecordLoungeVisit',
+        {},
+        {
+            optimisticData: [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: ONYXKEYS.USER,
+                    value: {loungeCheckInDetails: {isCheckedIn: true, checkInsRemaining: checkInsRemaining - 1}},
+                },
+            ],
+        },
+    );
```

Comment:
Should we set failure data as well? Otherwise, this will stay with `checkInsRemaining - 1`, right?

Reply:
Great point. Implemented.